### PR TITLE
Fix : Uniswap V3 Staker

### PR DESCRIPTION
**Pull request by bhavik-m**
- [x] Added function to check whether pool exist or not.
- [x] Deploy Resolver  https://github.com/Instadapp/dsa-resolvers/pull/78#issuecomment-1066789642
- [x] Update Deployed Resolver Address on defi-express-multichain Repo https://github.com/Instadapp/defi-express-multichain/pull/261

### CODE REVIEW

File: contracts/protocols/arbitrum/uniswapStaking/main.sol [hunk 1]
```diff
@@ -299,6 +299,17 @@ contract Resolver is Helpers {
...
+
+    function isPoolExist(address[] memory pools) public view returns (bool[] memory isValidPools) {
+        isValidPools = new bool[](pools.length);
+        for (uint256 i = 0; i < pools.length; i++) {
+            if (isContract(pools[i]) == true) {
+                isValidPools[i] = true;
+            } else {
+                isValidPools[i] = false;
+            }
+        }
```
Comment (pradyuman-verma) [inline contracts/protocols/arbitrum/uniswapStaking/main.sol]:
```suggestion
            isValidPools[i]  = isContract(pools[i]);
        }
```

File: contracts/protocols/polygon/uniswapStaking/main.sol [hunk 2]
```diff
@@ -299,6 +299,17 @@ contract Resolver is Helpers {
...
     }
+
+    function isPoolExist(address[] memory pools) public view returns (bool[] memory isValidPools) {
+        isValidPools = new bool[](pools.length);
+        for (uint256 i = 0; i < pools.length; i++) {
+            if (isContract(pools[i]) == true) {
+                isValidPools[i] = true;
+            } else {
+                isValidPools[i] = false;
+            }
```
Comment (pradyuman-verma) [inline contracts/protocols/polygon/uniswapStaking/main.sol]:
```suggestion
            isValidPools[i] =  isContract(pools[i]);
```

File: contracts/protocols/polygon/uniswapStaking/helpers.sol [hunk 3]
```diff
@@ -97,4 +97,13 @@ abstract contract Helpers is DSMath {
             );
         }
     }
+
+    function isContract(address account) public view returns (bool) {
+        uint32 size;
+        address a = account;
+        assembly {
+            size := extcodesize(a)
+        }
```
Comment (pradyuman-verma) [inline contracts/protocols/polygon/uniswapStaking/helpers.sol]: As we are using inline assembly, I think this function can't be of view type?
Can you try compiling the code?

File: contracts/protocols/polygon/uniswapStaking/helpers.sol [hunk 4]
```diff
@@ -97,4 +97,13 @@ abstract contract Helpers is DSMath {
             );
         }
     }
+
+    function isContract(address account) public view returns (bool) {
+        uint32 size;
+        address a = account;
```
Comment (pradyuman-verma) [inline contracts/protocols/polygon/uniswapStaking/helpers.sol]: Why declare again?

File: contracts/protocols/optimism/uniswapStaking/main.sol [hunk 5]
```diff
@@ -299,6 +299,17 @@ contract Resolver is Helpers {
...
     }
+
+    function isPoolExist(address[] memory pools) public view returns (bool[] memory isValidPools) {
+        isValidPools = new bool[](pools.length);
+        for (uint256 i = 0; i < pools.length; i++) {
+            if (isContract(pools[i]) == true) {
+                isValidPools[i] = true;
+            } else {
+                isValidPools[i] = false;
+            }
```
Comment (pradyuman-verma) [inline contracts/protocols/optimism/uniswapStaking/main.sol]:
```suggestion
           isValidPools[i] = isContract(pools[i]);
```

File: contracts/protocols/polygon/uniswapStaking/helpers.sol (hunk 3 above)
Comment (bhavik-m) [reply contracts/protocols/polygon/uniswapStaking/helpers.sol]: Yeah code is running fine

Comment (pradyuman-verma) [reply contracts/protocols/polygon/uniswapStaking/helpers.sol]: okay

**Comment by bhavik-m** [comment]
## Deployed Address
Polygon : [0xe9bacaabc08e9ccaa37237b72923b4227f21ed62](https://polygonscan.com/address/0xe9bacaabc08e9ccaa37237b72923b4227f21ed62)
Optimism : [0x54C624734F6f6b23e1c59FB0B2C3d0822C5A7135](https://optimistic.etherscan.io/address/0x54C624734F6f6b23e1c59FB0B2C3d0822C5A7135#code)
Mainnet : [0x9B5CA5D91F0C017Bd6f1Cb8543494Bd7DACC820d](https://etherscan.io/address/0x9b5ca5d91f0c017bd6f1cb8543494bd7dacc820d#code)
Arbitrum : [0x1310f478A37bBdccf47E8e10962d3bB888FEc62C](https://arbiscan.io/address/0x1310f478A37bBdccf47E8e10962d3bB888FEc62C#code)

File: hardhat.config.ts [hunk 6]
```diff
@@ -156,7 +164,7 @@ const config: HardhatUserConfig = {
     timeout: 10000 * 1000,
   },
   etherscan: {
-    apiKey: String(process.env.SCANAPI),
+    apiKey: getScanApiKey(String(process.env.networkType)),
```
Comment (pradyuman-verma) [inline hardhat.config.ts:167]: Let's update the hardhat etherscan version as well.

Comment (bhavik-m) [reply hardhat.config.ts:167]: Sure

**Pull request merged**